thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/HashService.js

```diff
@@ -42,11 +42,19 @@ HashService.prototype = {
     switch (key) {
       case HASH_PARAMS.ANOMALIES_START_DATE:
       case HASH_PARAMS.ANOMALIES_END_DATE:
+      case HASH_PARAMS.ANALYSIS_CURRENT_START:
+      case HASH_PARAMS.ANALYSIS_CURRENT_END:
+      case HASH_PARAMS.ANALYSIS_BASELINE_START:
+      case HASH_PARAMS.ANALYSIS_BASELINE_END:
         value = moment(value);
         break;
       case HASH_PARAMS.ANOMALIES_PAGE_NUMBER:
         value = Number(value);
         break;
+      case HASH_PARAMS.ANALYSIS_FILTERS:
+        if (typeof value === "string") {
```

Comment:
so the initial intention was that since this property can either be set from the controller as an object `{}` , or  taken from the hash params as a stringified object `"{}"`, I would parse it only in the latter case, the default value being `"{}"`
But I agree that a try catch might be cleaner.